pkgs/applications/networking/sniffers/wireshark/default.nix

```diff
@@ -23,7 +23,8 @@ in stdenv.mkDerivation {
     sha256 = "0bpiby916k3k8bm7q8b1dflva6zs0a4ircskrck0d538dfcrb50q";
   };
 
-  cmakeFlags = optional withGtk "-DBUILD_wireshark_gtk=TRUE";
+  cmakeFlags = (if withGtk then ["-DBUILD_wireshark_gtk=TRUE"] else ["-DBUILD_wireshark_gtk=OFF"]) ++
+               (if withQt then ["-DBUILD_wireshark=TRUE"] else ["-DBUILD_wireshark=OFF"]);
```

Comment:
It would probable be nicer to factor this out to a function, see for example

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/stxxl/default.nix#L6

Reply:
Including the use of `ON`/`OFF` instead of `TRUE`/`OFF`.

Reply:
With only two options inline is also fine, but it can be simplified to `cmakeFlags = [ "-DBUILD_wireshark_gtk=${if withGTK then "ON" else "OFF"}" … ]`.

Reply:
I still think [0–1–∞ rule](https://en.wikipedia.org/wiki/Zero_one_infinity_rule) results in more aesthetically pleasing code but this will work too.

Reply:
That rule is pretty, but such duplication as this one is slightly easier to reason about both for the author of the change and for the readers. (Because it is local and it does not introduce any new names or functions.)